### PR TITLE
[Operator] Add upsample_nearest2d op [MooreThreads]

### PR DESCRIPTION
**Pull request by ZaccurLi**
### PR Category

Operator `upsample_nearest2d`

### Type of Change

New Feature

### Description

https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml#L12777C6-L12777C7
func: upsample_nearest2d(Tensor self, SymInt[2] output_size, float? scales_h=None, float? scales_w=None) -> Tensor

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [ ] Change is fully covered by a UT.

### Performance

image

### CODE REVIEW

File: benchmark/test_special_perf.py [hunk 1]
```diff
@@ -140,6 +140,36 @@ def padding_kwargs(dtype, batch, size):
...
+            "input": input,
+            "output_size": output_size,
+            "scales_h": None,
+            "scales_w": None,
+        }
+
+    bench = Benchmark(
+        op_name="upsample_nearest2d",
+        # py api torch.nn.functional.interpolate,
+        torch_op=torch._C._nn.upsample_nearest2d,
```
Comment (Bowen12992) [inline benchmark/test_special_perf.py:163]: Can we just use the pytorch api torch.nn.functional.interpolate with proper args to make the api call the torch._C._nn._upsample_nearest2d functions

File: benchmark/test_special_perf.py [hunk 2]
```diff
@@ -140,6 +140,36 @@ def padding_kwargs(dtype, batch, size):
...
+        arg_func=None,
+        dtypes=[torch.float16, torch.float32],
```
Comment (Bowen12992) [inline benchmark/test_special_perf.py:165]: Is there any problem with bfloat16

**Review by Bowen12992 — COMMENTED** [review]
This PR has some common issue with PR https://github.com/FlagOpen/FlagGems/pull/203

**Comment by ZaccurLi** [comment]
PTAL @Bowen12992

**Review by Bowen12992 — APPROVED** [review]
LGTM

**Pull request merged**